migrations/000001_init_schema.up.sql

```diff
@@ -0,0 +1,32 @@
+CREATE TABLE IF NOT EXISTS users (
+    id uuid NOT NULL DEFAULT uuid_generate_v1(),
+    username varchar(512) NOT NULL UNIQUE,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS notification_types (
+    id UUID NOT NULL DEFAULT uuid_generate_v1(),
+    name varchar(32) NOT NULL UNIQUE,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS notifications (
+    id UUID DEFAULT uuid_generate_v1(),
+    notification_type_id uuid NOT NULL REFERENCES notification_types(id),
+    user_id uuid NOT NULL REFERENCES users(id) ON DELETE CASCADE,
+    subject text NOT NULL,
+    seen boolean NOT NULL DEFAULT false,
+    deleted boolean NOT NULL DEFAULT false,
+    time_created timestamp with time zone NOT NULL DEFAULT now(),
+    message jsonb NOT NULL,
+    PRIMARY KEY (id)
+);
+
+CREATE INDEX notifications_notification_type_id_index
```

Comment:
Looks like `CREATE INDEX ... IF NOT EXISTS` is supported since postgres 9.5. Not sure if we're potentially using anything older than that, but if not I suppose that'd be nice for idempotence.

Reply:
Ah, yes, that would be helpful. I'll add that.